addons/hr_work_entry_contract/tests/test_work_entry.py

```diff
@@ -259,3 +260,70 @@ def test_work_entry_employee_without_contract(self):
             work_entry.save()
         self.assertEqual(work_entry.employee_id.id, new_employee.id)
         self.assertEqual(work_entry.duration, 0.0)
+
+    def test_separate_overlapping_work_entries_by_type(self):
+        contract = self.env['hr.contract'].create({
+            'date_start': datetime(2024, 9, 1),
+            'date_end': datetime(2024, 9, 30),
+            'name': 'Contract',
+            'wage': 5000.0,
+        })
+
+        entry_type_1, entry_type_2 = self.env['hr.work.entry.type'].create([
+            {'name': 'Work type 1', 'is_leave': False, 'code': 'ENTRY_TYPE1'},
+            {'name': 'Work type 2', 'is_leave': False, 'code': 'ENTRY_TYPE2'},
+        ])
+
+        calendar = self.env['resource.calendar'].create({'name': 'Calendar', 'tz': 'Europe/Brussels'})
+
+        attendance_morning = self.env['resource.calendar.attendance'].create([
+            {
+                'calendar_id': calendar.id,
+                'dayofweek': '0',
+                'name': 'Same type 1',
+                'hour_from': 8,
+                'hour_to': 11,
+                'day_period': 'morning',
+                'work_entry_type_id': entry_type_1.id,
+            },
+            {
+                'calendar_id': calendar.id,
+                'dayofweek': '0',
+                'name': 'Same type 2',
+                'hour_from': 11,
+                'hour_to': 12,
+                'day_period': 'morning',
+                'work_entry_type_id': entry_type_1.id,
+            },
+        ])
+
+        attendance_afternoon = self.env['resource.calendar.attendance'].create([
+            {
+                'calendar_id': calendar.id,
+                'dayofweek': '0',
+                'name': 'Different types 1',
+                'hour_from': 13,
+                'hour_to': 16,
+                'day_period': 'afternoon',
+                'work_entry_type_id': entry_type_1.id,
+            },
+            {
+                'calendar_id': calendar.id,
+                'dayofweek': '0',
+                'name': 'Different types 2',
+                'hour_from': 16,
+                'hour_to': 17,
+                'day_period': 'afternoon',
+                'work_entry_type_id': entry_type_2.id,
+            },
+        ])
+
+        intervals = Intervals([
+            (datetime(2024, 9, 2, hour=8), datetime(2024, 9, 2, hour=12), attendance_morning),
+            (datetime(2024, 9, 2, hour=13), datetime(2024, 9, 2, hour=17), attendance_afternoon),
+        ])
+
+        result_intervals = contract._postprocess_attendance_intervals(intervals)
```

Comment:
just testing your method could be ok, but usually it's a lot more worth it to at least call the code that was failing (eg _get_contract_work_entries_values or a method above that in the call stack) because eg. if we remove the change in that method the test would still be green

Reply:
You are absolutely right, I changed the test to go through generate_work_entries() as suggested, that way it will go through the previously failing flow as well :+1: